src/components/MemoryViewer/MemoryViewer.tsx

```diff
@@ -0,0 +1,149 @@
+import {DefinitionList, useTheme} from '@gravity-ui/uikit';
+
+import type {TMemoryStats} from '../../types/api/nodes';
+import {cn} from '../../utils/cn';
+import {calculateProgressStatus} from '../../utils/progress';
+import {isNumeric} from '../../utils/utils';
+import {HoverPopup} from '../HoverPopup/HoverPopup';
+import {ProgressViewer} from '../ProgressViewer/ProgressViewer';
+
+import {getMemorySegments} from './utils';
+
+import './MemoryViewer.scss';
+
+const b = cn('memory-viewer');
+
+type FormatProgressViewerValues = (
+    value?: number,
+    capacity?: number,
+    precision?: number,
+) => (string | number | undefined)[];
+
+export interface MemoryProgressViewerProps {
+    stats: TMemoryStats;
+    className?: string;
+    warningThreshold?: number;
+    value?: number | string;
+    capacity?: number | string;
+    formatValues: FormatProgressViewerValues;
+    percents?: boolean;
+    dangerThreshold?: number;
+}
+
+const MEMORY_PRECISION = 2;
+
+export function MemoryViewer({
+    stats,
+    value,
+    capacity,
+    percents,
+    formatValues,
+    className,
+    warningThreshold = 60,
+    dangerThreshold = 80,
+}: MemoryProgressViewerProps) {
+    const theme = useTheme();
+    let fillWidth =
+        Math.round((parseFloat(String(value)) / parseFloat(String(capacity))) * 100) || 0;
+    fillWidth = fillWidth > 100 ? 100 : fillWidth;
+    let valueText: number | string | undefined = value,
+        capacityText: number | string | undefined = capacity,
+        divider = '/';
+    if (percents) {
+        valueText = fillWidth + '%';
+        capacityText = '';
+        divider = '';
+    } else if (formatValues) {
+        [valueText, capacityText] = formatValues(Number(value), Number(capacity), MEMORY_PRECISION);
+    }
+
+    const renderContent = () => {
+        if (isNumeric(capacity)) {
+            return `${valueText} ${divider} ${capacityText}`;
+        }
+
+        return valueText;
+    };
+
+    const calculateMemoryShare = (segmentSize: number) => {
+        if (!value) {
+            return 0;
+        }
+        return (segmentSize / parseFloat(String(capacity))) * 100;
+    };
+
+    const memorySegments = getMemorySegments(stats);
+
+    const totalUsedMemory =
```

Comment:
1. It's actually not total used memory, but it's memory usage in percents since you divide by capacity
2. Why don't you use `totalUsedMemory` (or `fillWidth`) everywhere?

Reply:
2 - what do you mean by "use everywhere" ?

Reply:
You calculated `fillWidth` and then you calculate `totalUsedMemory`, they should be the same. I mean not to calculate two values, but use single value instead, that is calculated once